### PR TITLE
Upgrade to Elasticsearch 7.10 (#3006)

### PR DESCRIPTION
**Pull request by hannes-ucsc**
https://github.com/DataBiosphere/azul/issues/3006

Announcement before sandbox build:

> The sandbox ES instance will be destroyed and recreated. If your personal deployments share the ES instance with the sandbox, their indices will be lost. Hold off using such personal deployments until further notice.

Announcement after merge (please update link to point to merge commit on develop]:

> The sandbox ES instance has been recreated and uses ElasticSearch 7.10. Please rebase your feature branches and [upgrade](https://github.com/DataBiosphere/azul/blob/01b5ba2933938d223fecdd542cd81966ee9932f2/UPGRADING.rst#3006-upgrade-to-elasticsearch-710) your personal deployments, regardless of whether your personal deployments share the sandbox ES instance or not.

### Author

- [x] PR title references issue
- [x] PR title matches issue title (preceded by `Fix: ` for bugs)   <sub>or there is a good reason why they're different</sub>
- [x] Title of main commit references issue
- [x] PR is connected to Zenhub issue and description links to issue

### Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

### Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>
- [x] Added announcement to PR description                  <sub>or this PR does not require announcement</sub>
- [x] Added checklist items for additional operator tasks   <sub>or this PR does not require additional tasks</sub>

### Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR does not touch requirements*.txt</sub>

### Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

### Primary reviewer (after approval)

- [x] Commented in issue about demo expectations            <sub>or labelled issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear              <sub>or issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Made announcement requested by author                 <sub>or PR description does not contain an announcement</sub>
- [x] Deleted the `azul-index-sandbox` ES domain
- [x] Branch pushed to Gitlab and added `sandbox` label     <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passed in sandbox                               <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

### Operator (after pushing the merge commit)

- [ ] Made announcement requested by author                 <sub>or PR description does not contain an announcement</sub>
- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate       <sub>or this PR is authored by lead</sub>
- [ ] Deleted the `azul-index-dev` ES domain
- [ ] Pushed merge commit to Gitlab                         <sub>or merge commit can be pushed later, with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Build passes on Gitlab
- [ ] Moved issues to `prod` or `Merged prod`               <sub>or this PR does not represent a promotion</sub>

### Operator (reindex) 

- [ ] Started reindex in target deployment                  <sub>or this PR does not require reindexing</sub>
- [ ] Checked for and triaged indexing failures             <sub>or this PR does not require reindexing</sub>
- [ ] Emptied fail queues in target deployment              <sub>or this PR does not require reindexing</sub>
- [ ] Filed backport PR                                     <sub>or this PR does not represent a hotfix</sub>

### Operator

- [ ] Unassigned PR

### CODE REVIEW

File: terraform/elasticsearch.tf.json.template.py [hunk 1]
```diff
@@ -94,8 +94,10 @@
                         "ebs_enabled": "true",
```
Comment (amarjandu) [inline terraform/elasticsearch.tf.json.template.py:94]: Do these ebs_options need to be removed off? 

The R6GD instance types require Elasticsearch 7.9 or later or any version of OpenSearch and do not support EBS storage volumes.

https://docs.aws.amazon.com/opensearch-service/latest/developerguide/supported-instance-types.html

Comment (hannes-ucsc) [reply terraform/elasticsearch.tf.json.template.py:94]: The conditional below should take care of that.

**Comment by hannes-ucsc** [comment]
Sandbox build failed with https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/37687#L7463

**Comment by hannes-ucsc** [comment]
I obviously tried this in my personal deployment (with its own ES domain), which is why I am surprised that the deployment of sandbox failed. I tested some more scenarios today, and I can create my deployment's ES domain and I can change the domain's instance types, all with the old `.elasticsearch` type names. I think I'll just update the checklist to just manually delete the ES domain in the target deployment.

**Comment by hannes-ucsc** [comment]
IT failed in `dev`: https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/37819

https://github.com/DataBiosphere/azul/issues/3786

`dev` is the first deployment with more than 25k files, exposing that bug.

**Comment by hannes-ucsc** [comment]
IT passes with [fix](https://github.com/DataBiosphere/azul/commit/a7a5abf1fdd295547cca1d13ffdd37e8d40d2f15) for #3786

https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/pipelines/11021

**Pull request merged**